packages/nimble-blazor/Tests/NimbleBlazor.Tests/Unit/Components/NimbleTableTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Text.Json;
+using Bunit;
+using Xunit;
+#nullable enable
+namespace NimbleBlazor.Tests.Unit.Components;
+
+/// <summary>
+/// Tests for <see cref="NimbleTable"/>
+/// </summary>
+public class NimbleTableTests
+{
+    internal class TableRecord
+    {
+        public TableRecord(string firstName, string lastName)
+        {
+            FirstName = firstName;
+            LastName = lastName;
+        }
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+    }
+
+    internal class BadTableRecord
+    {
+        public BadTableRecord(string firstName, string lastName, BadTableRecord? parent = null)
+        {
+            FirstName = firstName;
+            LastName = lastName;
+            Parent = parent;
+        }
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public BadTableRecord? Parent { get; set; }
+    }
+
+    [Fact]
+    public void NimbleTable_Rendered_HasTableMarkup()
+    {
+        var context = new TestContext();
+        context.JSInterop.Mode = JSRuntimeMode.Loose;
+        var expectedMarkup = "nimble-table";
+        var table = context.RenderComponent<NimbleTable<TableRecord>>();
+
+        Assert.Contains(expectedMarkup, table.Markup);
+    }
+
+    [Fact]
+    public void NimbleTable_GivenSupportedData_ThrowsNoException()
+    {
+        var parentRecord = new TableRecord("Bob", "Smith");
+        var childRecord = new TableRecord("Sally", "Smith");
+        var tableData = new TableRecord[] { parentRecord, childRecord };
+
+        var ex = Record.Exception(() => { RenderWithPropertySet<IEnumerable<TableRecord>, TableRecord>(x => x.Data, tableData); });
```

Comment:
nit: it took me a bit to understand what `Record` is here. I understand now that it's a test utility from `Xunit` and it refers to the verb "record" (i.e. record any exceptions that happen while executing this code)? My confusion came from the fact that this code also uses the noun "record" extensively (e.g. `TableRecord`).

Some ideas to make this more clear:
1. alias `Xunit.Record` to use a different name via a [using alias directive](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-directive)
2. don't include `using Xunit;` and rather use its fully qualified name (`Xunit.Record.Exception(...)`) everywhere in the file 
3. use a different test structure instead of `Record`. For example, a `try`/`catch` with a failing assert in the `catch`. (We probably have test conventions that tell us what to do and this my violate those)

I could also be convinced to leave it as is if the above turn out to be ugly in practice.

Reply:
I think I would just prefer to rename the local test classes to something like `TableRowData` and `BadTableRowData`. Ultimately, I think using `Record(...)` is fine, as is, despite the confusion when coming across it in a code review. In practice, a developer will have intellisense to provide the context for what that method is, which will make its purpose obvious and clear.

Reply:
That would work for me.

Reply:
Done.